helm/external-dns-app/templates/deployment.yaml

```diff
@@ -11,12 +11,18 @@ spec:
   selector:
     matchLabels:
       {{- include "labels.selector" . | nindent 6 }}
+      {{- if and (eq .Values.provider "gcp") (.Values.gcpServiceAccountID) (.Values.gcpWorkloadIdentityPool) (.Values.gcpIdentityProvider) }}
+      giantswarm.io/gcp-workload-identity: enabled
+      {{- end }}
   strategy:
     type: Recreate
   template:
     metadata:
       labels:
         {{- include "labels.common" . | nindent 8 }}
+        {{- if and (eq .Values.provider "gcp") (.Values.gcpServiceAccountID) (.Values.gcpWorkloadIdentityPool) (.Values.gcpIdentityProvider) }}
+        giantswarm.io/gcp-workload-identity: enabled
```

Comment:
Rather move this to the `labels.common` template.

Reply:
I don't want the label on any other resource, only on the pods.